diwan/src/commands/file_commands.rs

```diff
@@ -0,0 +1,24 @@
+pub enum CommandType {
+    Open,
+    Save,
+    SaveAs,
+}
+
+pub struct FileCommand {
+    command_type: CommandType,
+    // Command-related fields
+}
+
+impl FileCommand {
+    pub async fn handle_open(&self, file_name: &str) -> std::io::Result<String> {
```

Comment:
```suggestion
    pub async fn handle_open(&self, file_name: &Path) -> std::io::Result<String> {
```
It is recommended to take a path as an argument instead of a string. This allows you to perform operations such as checking if the file exists (e.g., if `filename.exists()`) and handle other path-related tasks more effectively.